draft-ietf-ppm-dap.md

```diff
@@ -1501,11 +1512,15 @@ its query:
 
 ### Batch Validation {#batch-validation}
 
-Before an Aggregator responds to a CollectReq or AggregateShareReq, it must
+Before an Aggregator responds to a CollectReq or AggregateShareReq, it MUST
 first check that the request does not violate the parameters associated with the
 DAP task. It does so as described here.
 
-First the Aggregator checks that the batch respects any "boundaries" determined
+First the Aggregator checks that the task still exists, if the task identified by
+`task_id` no longer exists (for e.g. deleted after `max_task_lifetime` passed),
+then the Aggregator MUST abort with error "unrecognizedTask".
```

Comment:
I don't think we need this paragraph, but for a different reason: it should be OK to do aggregations on a task after the aggregators have stopped accepting client reports for it. Let's say a task has `max_task_lifetime` `t` and it takes `d` seconds to run the collect protocol. So far as clients know, it is OK for them to continue submitting reports right up until instant `t`. As written, the collector must initiate the collect protocol at time `t - d`. So that means any reports that arrive during the interval `(t - d, t)` won't be included in the aggregation.

The other problem is that there's no good way in the protocol for the collector to figure out what `d` is and thus figure out how late it can issue a collect request without risking losing an aggregation because the task expired.

In fact, that seems to be implied by the text added in "Reducing storage requirements".

Reply:
I'll revert the change.

@tgeoghegan The current description indeed doesn't say explicitly how soon a task should be retired after expiry, IMO it should be left to the deployment. Collector should know `t` and try to collect before `t` expires.